### DISCUSSION TITLE
git repo initialization directive

### DISCUSSION THREAD

**Discussion by larssb** [PowerShellOrg/Plaster#391] [Ideas]
This is a follow up to comments in #5 from 170623.

- It would be great to have a directive available in Plaster that could initialize the project as a git repository. This would help users of my Plaster template to get going even faster. More stuff ready more happy users.

Thank you.

**Comment by rkeithhill** [comment]
I'd like to use [libgit2sharp](https://github.com/libgit2/libgit2sharp) for this so we don't have to rely on Git being in the path.  We will need to redist the libgit2sharp assembly.  @daviwil the [license for the project is MIT](https://github.com/libgit2/libgit2sharp/blob/master/LICENSE.md), will this be a problem?

BTW it looks like this project will be [adding support for .NET Core shortly](https://github.com/libgit2/libgit2sharp/releases/tag/v0.24).  I'll wait for them to get their .NET Core support in place.  Hopefully that won't be too long.

**Comment by larssb** [comment]
@rkeithhill reads like a total great idea :-).

**Comment by jdhitsolutions** [comment]
Any progress on this? Right now I have git-related logic in a controller script.
